TestsCalculator/SettingsPageTests.cs

```diff
@@ -0,0 +1,199 @@
+
+            //Act
+            SelectElement s = settingsPage.DateFormatOptions;
+
+            IList<string> actualDateFormat = new List<string>();
+            for (int j = 0; j < s.Options.Count; j++)
+            {
+                actualDateFormat.Add(s.Options.ElementAt(j).Text);
+            }
+            //Assert
+            Assert.AreEqual(expectedDateFormat, actualDateFormat);
+        }
+
+        [Test]
+        public void CheckNumberFormatOptions()
+        {
+            //Arrange
+            SettingsPage settingsPage = new SettingsPage(driver);
+            List<string> expectedNumberFormat = new List<string> { "123,456,789.00", "123.456.789,00", "123 456 789.00", "123 456 789,00" };
+
+            //Act
+            SelectElement s = settingsPage.NumberFormatOptions;
+            IList<string> actualNumberFormat = new List<string>();
+            for (int j = 0; j < s.Options.Count; j++)
+            {
+                actualNumberFormat.Add(s.Options.ElementAt(j).Text);
+            }
+
+            //Assert
+            Assert.AreEqual(expectedNumberFormat, actualNumberFormat);
+        }
+
+        [Test]
+        public void CheckCurrencyFormatOptions()
+        {
+            //Arrange
+            SettingsPage settingsPage = new SettingsPage(driver);
+            List<string> expectedCurrencyFormat = new List<string> { "$ - US dollar", "€ - euro", "£ - Great Britain Pound" };
+
+            //Act
+            SelectElement s = settingsPage.CurrencyFormatOptions;
+            IList<string> actualCurrencyFormat = new List<string>();
+            for (int j = 0; j < s.Options.Count; j++)
+            {
+                actualCurrencyFormat.Add(s.Options.ElementAt(j).Text);
+            }
+
+            //Assert
+            Assert.AreEqual(expectedCurrencyFormat, actualCurrencyFormat);
+        }
+
+        [Test]
+        public void CancelBtnOpensDepositPage()
+        {
+            //Arrange
+            SettingsPage settingsPage = new SettingsPage(driver);
+
+            //Act
+            settingsPage.CancelBtn.Click();
+
+            //Assert
+            string actualUrl = driver.Url;
+            string expectedUrl = "http://127.0.0.1:8080/Deposit";
+            Assert.AreEqual(expectedUrl, actualUrl);
+        }
+
+        [TestCase("MM dd yyyy", "M dd yyyy")]
+        [TestCase("MM/dd/yyyy", "M/dd/yyyy")]
+        [TestCase("dd-MM-yyyy", "dd-M-yyyy")]
+        [TestCase("dd/MM/yyyy", "dd/M/yyyy")]
+        public void SaveDateFormat(string dateFormat, string dateDisplay)
+        {
+            //Arrange
+            SettingsPage settingsPage = new SettingsPage(driver);
+            CalculatorPage calculatorPage = new CalculatorPage(driver);
+
+            //Act
+            settingsPage.DateFormatOptions.SelectByText(dateFormat);
+            settingsPage.SaveBtn.Click();
+            IAlert alert = driver.SwitchTo().Alert();
+            alert.Accept();
```

Comment:
No 'driver' is allowed in tests 😈
How to handle it? Create a method Save in Settings page:
```
public void Save()
{
    SaveBtn.Click();
    driver.SwitchTo().Alert().Accept();
}
```

Reply:
corrected